src/modules/accounting/accounting.py

```diff
@@ -1,31 +1,169 @@
+            if not self.is_extra_data_submitted(latest_blockstamp):
+                self._submit_extra_data(report_timestamp)
+
+    def _submit_extra_data(self, blockstamp: BlockStamp) -> None:
+        extra_data = self.lido_validator_state_service.get_extra_data(blockstamp)
+
+        tx = self.report_contract.functions.submitReportExtraDataList(extra_data.extra_data)
+
+        if self.w3.transaction.check_transaction(tx, variables.ACCOUNT.address):
+            self.w3.transaction.sign_and_send_transaction(tx, variables.GAS_LIMIT, variables.ACCOUNT)
+
+    # Consensus module: main build report method
+    @lru_cache(maxsize=1)
+    def build_report(self, blockstamp: BlockStamp) -> tuple:
+        logger.info({'msg': 'Calculate report for accounting module.'})
+        report_data = self._calculate_report(blockstamp)
+        return report_data.as_tuple()
+
+    # # Consensus module: if contract got report data
+    def is_main_data_submitted(self, blockstamp: BlockStamp) -> bool:
+        processing_state = self._get_processing_state(blockstamp)
+        logger.info({'msg': 'Check if main data was submitted.', 'value': processing_state.main_data_submitted})
+        return processing_state.main_data_submitted
+
+    # Consensus module: if contract could accept any sort of report
+    def is_contract_reportable(self, blockstamp: BlockStamp) -> bool:
+        is_reportable = not self.is_main_data_submitted(blockstamp) or not self.is_extra_data_submitted(blockstamp)
+        logger.info({'msg': 'Check if contract could accept report.', 'value': is_reportable})
+        return is_reportable
+
+    @lru_cache(maxsize=1)
+    def _get_processing_state(self, blockstamp: BlockStamp) -> ProcessingState:
+        ps = named_tuple_to_dataclass(
+            self.report_contract.functions.getProcessingState().call(block_identifier=blockstamp.block_hash),
+            ProcessingState,
+        )
+        logger.info({'msg': 'Fetch processing state.', 'value': ps})
+        return ps
+
+    def is_extra_data_submitted(self, blockstamp: BlockStamp) -> bool:
+        processing_state = self._get_processing_state(blockstamp)
+        return processing_state.extra_data_items_count == processing_state.extra_data_items_submitted
+
+    def _calculate_report(self, blockstamp: BlockStamp) -> ReportData:
+        validators_count, cl_balance = self._get_consensus_lido_state(blockstamp)
+
+        finalization_share_rate = self._get_finalization_shares_rate(blockstamp)
+        last_withdrawal_id_to_finalize = self._get_last_withdrawal_request_to_finalize(blockstamp)
+
+        exited_validators = self.lido_validator_state_service.get_lido_new_exited_validators(blockstamp)
+
+        # Here report all exited validators even they were reported before.
+        if exited_validators:
+            stacking_module_id_list, exit_validators_count = zip(*exited_validators.items())
+        else:
+            stacking_module_id_list = exit_validators_count = []
```

Comment:
I would suggest to think about naming for `stacking_module_id_list`, `exit_validators_count` because the names do not represent what they are supposed to contain